From e81dde14d9f9c98bdce4cc5536b105a6adc3613b Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Sun, 1 Sep 2024 14:24:08 +0200
Subject: [PATCH] [krdb] Don't manually process events when applying Qt
 settings

Manually processing events like this can cause unpredictable behavior and for QML apps like systemsettings cause crashes

It's also not necessary, the config we are applying isn't used by systemsettings/modern Qt apps anyway

BUG: 486922

SENTRY: SYSTEMSETTINGS-3RP
(cherry picked from commit aaf0cc601a7a9ed455e2a61906ed46b6e275f007)
---
 kcms/krdb/krdb.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/kcms/krdb/krdb.cpp b/kcms/krdb/krdb.cpp
index eefa0d36a0..39bcb7317d 100644
--- a/kcms/krdb/krdb.cpp
+++ b/kcms/krdb/krdb.cpp
@@ -484,7 +484,6 @@ void runRdb(unsigned int flags)
             applyQtSettings(kglobalcfg, *settings); // For kcmstyle
 
         delete settings;
-        QCoreApplication::processEvents();
 #if HAVE_X11
         if (qApp->platformName() == QLatin1String("xcb")) {
             // We let KIPC take care of ourselves, as we are in a KDE app with
@@ -514,7 +513,6 @@ void runRdb(unsigned int flags)
                             PropModeReplace,
                             (unsigned char *)stamp.buffer().data(),
                             stamp.buffer().size());
-            qApp->processEvents();
         }
 #endif
     }
-- 
GitLab

